backend/pkg/k8cache/k8cache.go

```diff
@@ -534,3 +534,83 @@ func CheckForChanges(k8scache cache.Cache[string],
 	}
 	mu.Unlock()
 }
+
+// Details provides information about a specific resource kind.
+type Details struct {
+	Kind string `json:"kind"` // The resource Kind ie, Pods, Nodes etc.
+}
+
+// Metadata holds core data about a resource, such as its version.
+type MetaData struct {
+	ResourceVersion string `json:"resourceVersion"`
+}
+
+// AuthErrResponse is the Unauthorized Error message that can be used
+// for sending 403 or Unauthorized error when the user is not allowed
+// to access resources.
+type AuthErrResponse struct {
+	Kind       string   `json:"kind"`       // The Kubernetes resource kind.
+	APIVersion string   `json:"apiVersion"` // APIVersion is version for the resource.
+	MetaData   MetaData `json:"metadata"`   // Metadata for the resource.
+	Message    string   `json:"message"`    // A human-readable error message.
+	Reason     string   `json:"reason"`     // Reason for the error.
+	Details    Details  `json:"details"`    // Details about the resource kind.
+	Code       int      `json:"code"`       // The HTTP status code, typically 403.
+}
+
+// CheckForURL help to identify the url(s) which should not passed while
+// returning Unauthorized Error to the client.
+func CheckForURL(urlPath string) bool {
```

Comment:
Hmm. CheckForURL doesn't really describe it...

Can you please try a different name?

ShouldReturnAuthErrorForURL(urlPath) is probably too long?
Maybe AllowAuthError(urlPath) ?